repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java

```diff
@@ -4713,28 +4718,35 @@ private void updateDomainAttribute(AtlasVertex vertex, Set<String> existingValue
         existingValues.forEach(existingValue -> vertex.removePropertyValue(DOMAIN_GUIDS_ATTR, existingValue));
         vertex.setProperty(DOMAIN_GUIDS_ATTR, meshEntityId);
     }
-    public AtlasVertex moveBusinessPolicy(Set<String> policyIds, String assetId, String type) throws AtlasBaseException {
+    public AtlasVertex moveBusinessPolicies(Set<String> policyIds, String assetId, String type) throws AtlasBaseException {
         // Retrieve the AtlasVertex for the given assetId
         AtlasVertex assetVertex = AtlasGraphUtilsV2.findByGuid(graph, assetId);
 
-        if(assetVertex == null){
+        if (assetVertex == null) {
             throw new AtlasBaseException(AtlasErrorCode.INVALID_PARAMETERS, "assetId not found");
         }
 
         // Get the sets of governed and non-compliant policy GUIDs
         Set<String> governedPolicies = assetVertex.getMultiValuedSetProperty(ASSET_POLICY_GUIDS, String.class);
         Set<String> nonCompliantPolicies = assetVertex.getMultiValuedSetProperty(NON_COMPLIANT_ASSET_POLICY_GUIDS, String.class);
 
-        // Determine if the type is governed or non-compliant and move policies accordingly
+        // Determine if the type is governed or non-compliant
         boolean isGoverned = MoveBusinessPolicyRequest.Type.GOVERNED.getDescription().equals(type);
+        Set<String> currentPolicies = isGoverned ? governedPolicies : nonCompliantPolicies;
 
+        // Check if the asset already has the given policy IDs
+        if (currentPolicies.equals(policyIds)) {
+            throw new AtlasBaseException(AtlasErrorCode.BAD_REQUEST, "Asset already has the given policy id");
+        }
+
+        // Move policies to the appropriate set
         policyIds.forEach(policyId -> {
             if (isGoverned) {
                 assetVertex.setProperty(ASSET_POLICY_GUIDS, policyId);
-                AtlasGraphUtilsV2.removeItemFromListPropertyValue(assetVertex, NON_COMPLIANT_ASSET_POLICY_GUIDS, policyId);
+                removeItemFromListPropertyValue(assetVertex, NON_COMPLIANT_ASSET_POLICY_GUIDS, policyId);
```

Comment:
Imagine we sent policyIds that are not at all linked to Asset, neither governed nor non_compliant.
Please confirm if this can be the case or if WF handles this use case & metastore need not worry about this validation